repotracker/repotracker.go

```diff
@@ -647,11 +646,14 @@ func CreateVersionFromConfig(ctx context.Context, projectInfo *model.ProjectInfo
 			v.Errors = append(v.Errors, versionErrs.Errors...)
 		}
 		if len(v.Errors) > 0 {
-			if err = v.Insert(); err != nil {
-				return nil, errors.Wrap(err, "error inserting version")
+			env := evergreen.GetEnvironment()
```

Comment:
nit: would it make sense to pass down the env from the repotrackerJob like we do with the context?

Reply:
I would love to do that, but unfortunately, `CreateVersionFromConfig` is used in a lot of places, of which some of them are really deep in the task lifecycle, so propagating the environment down from the top level for all of them would be a large amount of of refactoring of task lifecycle functions.

Reply:
that makes sense!